packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -94,63 +193,57 @@ export abstract class OptimisticOracleClient<R extends OptimisticOracleRequest>
    * @returns A new instance of OptimisticOracleClient.
    */
   protected abstract createClientInstance(
-    requests: R[],
-    fetchedBlockRanges: [number, number][]
+    requests: Map<string, R>,
+    fetchedBlockRange: BlockRange
   ): OptimisticOracleClient<R>;
 
   /**
-   * Retrieves Optimistic Oracle requests within the specified block range.
-   * @param blockRange The block range to fetch requests from.
-   * @returns A Promise that resolves to an array of OptimisticOracleRequest objects.
+   * Updates the OptimisticOracleClient instance by fetching new Oracle requests updates and storing them in the requests map.
+   * @param blockRange The new blockRange to fetch requests from.
    */
-  protected abstract fetchOracleRequests(blockRange: [number, number]): Promise<R[]>;
+  protected abstract updateOracleRequests(blockRange: BlockRange): Promise<void>;
 
   /**
    * Updates the OptimisticOracleClient instance by fetching new Oracle requests within the specified block range. Returns a new instance.
    * @param blockRange (Optional) The block range to fetch new requests from.
    * @returns A Promise that resolves to a new OptimisticOracleClient instance with updated requests.
    */
-  async updateWithBlockRange(blockRange?: [number, number]): Promise<this> {
-    let range: [number, number];
+  async updateWithBlockRange(blockRange?: BlockRange): Promise<this> {
```

Comment:
Would this avoid the cast at the bottom?
```suggestion
  async updateWithBlockRange(blockRange?: BlockRange): Promise<OptimisticOracleClient<R>> {
```